test/testdata/resolver/type_alias_self_type.rb

```diff
@@ -0,0 +1,27 @@
+# typed: strict
+
+class Example
+  extend T::Sig
+
+  # This magically becomes "the selftype of wherever this type alias is used"
+  # instead of "the self type of `self` right here" (i.e., the singleton class)
+  # I'm not sure if this behavior is intentional or accidental, but it is what
+  # it is so now we have a test for it. At least we don't crash.
```

Comment:
Great comment, thank you.  Wish this could be documented in the code somewhere if it's not already